src/main/resources/assets/computercraft/lua/rom/programs/shell

```diff
@@ -180,7 +180,11 @@ function shell.programs( _bIncludeHidden )
 	-- Sort and return
 	local tItemList = {}
 	for sItem, b in pairs( tItems ) do
-		table.insert( tItemList, sItem )
+	    if sItem:find(".lua") == sItem:len()-3 then
```

Comment:
The above section already ensures items are only collected once, so you just need to check that the file without `.lua` isn't also in `tItems`.